cmake/qpoases_interface/qpoases_interface.cpp

```diff
@@ -30,26 +33,88 @@ This file is part of DQ Robotics.
 
 using namespace DQ_robotics;
 
-int main(void)
-{
+void test_solver(void) {
     DQ_QPOASESSolver qpoases_solver;
     DQ_SerialManipulatorDH robot = KukaLw4Robot::kinematics();
 
-    DQ_ClassicQPController classic_qp_controller(&robot,&qpoases_solver);
+    DQ_ClassicQPController classic_qp_controller(&robot, &qpoases_solver);
     classic_qp_controller.set_control_objective(ControlObjective::Pose);
     classic_qp_controller.set_gain(0.01);
     classic_qp_controller.set_damping(0.01);
 
     VectorXd theta_init(7);
-    theta_init << 0.,pi/4.,0.,0.,pi/4.,0.,0.;
+    theta_init << 0., pi / 4., 0., 0., pi / 4., 0., 0.;
 
     VectorXd theta_xd(7);
-    theta_xd << 0.,pi/2.,0.,0.,pi/2.,0.,0.;
+    theta_xd << 0., pi / 2., 0., 0., pi / 2., 0., 0.;
     DQ xd = robot.fkm(theta_xd);
 
-    VectorXd qd = classic_qp_controller.compute_setpoint_control_signal(theta_init,vec8(xd));
+    VectorXd qd = classic_qp_controller.compute_setpoint_control_signal(theta_init, vec8(xd));
 
     std::cout << "q_dot is " << qd.transpose() << std::endl;
+}
+
+
+bool are_vectors_equal(const Eigen::VectorXd &A, const Eigen::VectorXd &B, double tolerance = 1e-3) {
+    if (A.size() != B.size()) {
+        return false; // Vectors have different dimensions
+    }
+
+    for (int i = 0; i < A.size(); ++i) {
+        if (std::abs(A(i) - B(i)) > tolerance) {
+            std::cout << "CHECK::index: " << i << " exceed error: " << std::abs(A(i) - B(i)) << std::endl;
+            return false; // Elements at (i) differ by more than the tolerance
+        }
+    }
+
+    return true;
+}
+
+void test_equality(void) {
+    DQ_QPOASESSolver qpoases_solver;
+    auto tolerance = qpoases_solver.get_equality_constraints_tolerance();
+//        tolerance = 0.001;
+    std::cout << "current equality tolerance: " << tolerance << std::endl;
+
+    for (int problem_size = 5; problem_size <= 10; problem_size += 1) {
+        qpoases_solver = DQ_QPOASESSolver();
+        tolerance = qpoases_solver.get_equality_constraints_tolerance();
```

Comment:
@qlin960618 It looks like you are using the same tolerance for all tests. If that is the case, please remove from the `for` loop the setter and getter lines related to the tolerance. If you want to test both methods, you could put them outside the `for` loop. For instance:

```cpp
//....
void test_equality(void) {
    DQ_QPOASESSolver qpoases_solver;
    auto tolerance = qpoases_solver.get_equality_constraints_tolerance();
    qpoases_solver.set_equality_constraints_tolerance(tolerance);
    std::cout << "current equality tolerance: " << tolerance << std::endl;

    for (int problem_size = 5; problem_size <= 10; problem_size += 1) {
        qpoases_solver = DQ_QPOASESSolver();
        MatrixXd H = MatrixXd::Identity(problem_size, problem_size);
        VectorXd f = VectorXd::Ones(problem_size);
//...
```